sycl/test-e2e/KernelQueries/test_plan.md

```diff
@@ -0,0 +1,53 @@
+# Test plan for [`sycl ext intel kernel queries`][spec-link] extension
+
+## Testing scope
+
+### Device coverage
+
+The unit tests should be launched on every supported device configuration we 
+have.
+
+### Type coverage
+
+Each query is templated on a single template type argument `Param`
+with some queries being template overloads of each other.
+Each query has a restriction on the value of `Param` defined
+in the relevant section of the spec.
+Each `Param` defines a related sycl aspect that signals whether the device
+supports it.
+
+Param must be one of the following types defined in 
+`sycl::ext::intel::info::kernel_device_specific` namespace:
+- `spill_memory_size`
+
+The tests should cover all of these types.
+
+## Tests
+
+### Unit tests
```

Comment:
Sorry for delayed review. Test plan indicates that unit tests are going to be added (i.e. https://github.com/intel/llvm/tree/sycl/sycl/unittests) but corresponding  PR - https://github.com/intel/llvm/pull/17594 - actually adds e2e lit test, not a unit test.

So, please change the wording in the test plan.

Reply:
I recently discovered that that test won't be of any use. And if I add the requirement on the aspect the test would be actually dropped. I'll try to add a unittest this week that'd use mock ur.

Reply:
That sounds good, in that case this test plan looks good, but should it be moved to unittests directory?

Reply:
Sure, done